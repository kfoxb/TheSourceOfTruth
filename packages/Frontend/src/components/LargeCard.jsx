import React, { Component } from 'react';
import { Card, Icon, Image } from 'semantic-ui-react';
import SealedPortion from './assets/SealedPortion.jpg';


export default class LargeCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      starFilled: false,
    };
  }
  toggleFav = () => this.setState({ starFilled: !this.state.starFilled })
  render() {
    const { starFilled } = this.state;
    return (
      <div style={{ position: 'relative', display: 'inline' }}>
        <Card>
          <Icon
            name="star"
            color={starFilled ? 'yellow' : 'whitesmoke'}
            onClick={this.toggleFav}
            size="huge"
            style={{
              color: 'whitesmoke',
              position: 'absolute',
              zIndex: 1,
              right: 0,
              top: 9,
            }}
          />
          <Image src={SealedPortion} />
          <Card.Content>
            <Card.Header>
              The Sealed Portion
            </Card.Header>
          </Card.Content>
          <Card.Content extra>
            <p>
              <Icon name="file text outline" />
              22 hr read
            </p>
          </Card.Content>
        </Card>
      </div>
    );
  }
}
