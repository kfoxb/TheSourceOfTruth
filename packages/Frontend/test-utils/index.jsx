/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
