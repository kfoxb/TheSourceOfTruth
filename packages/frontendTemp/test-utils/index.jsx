/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
