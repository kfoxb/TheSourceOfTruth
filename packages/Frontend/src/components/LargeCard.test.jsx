/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import LargeCard from './LargeCard';

describe('LargeCard', () => {
  const defaultProps = {
    starFilled: false,
  };
  const assertWithPropsToMatchSnapshot =
    createAssertWithPropsToMatchSnapshot(LargeCard, defaultProps);
  it('should render', () => {
    assertWithPropsToMatchSnapshot();
  });

  it('should render as star filled when passed starFilled as a prop', () => {
    assertWithPropsToMatchSnapshot({ starFilled: true });
  });
});
