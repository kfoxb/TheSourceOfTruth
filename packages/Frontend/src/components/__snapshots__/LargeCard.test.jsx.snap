// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeCard should render 1`] = `
<div
  style={
    Object {
      "display": "inline",
      "position": "relative",
    }
  }
>
  <Card>
    <Icon
      as="i"
      color="whitesmoke"
      name="star"
      onClick={[Function]}
      size="huge"
      style={
        Object {
          "color": "whitesmoke",
          "position": "absolute",
          "right": 0,
          "top": 9,
          "zIndex": 1,
        }
      }
    />
    <Image
      as="img"
      ui={true}
    />
    <CardContent>
      <CardHeader>
        The Sealed Portion
      </CardHeader>
    </CardContent>
    <CardContent
      extra={true}
    >
      <p>
        <Icon
          as="i"
          name="file text outline"
        />
        22 hr read
      </p>
    </CardContent>
  </Card>
</div>
`;

exports[`LargeCard should render as star filled when passed starFilled as a prop 1`] = `
<div
  style={
    Object {
      "display": "inline",
      "position": "relative",
    }
  }
>
  <Card>
    <Icon
      as="i"
      color="whitesmoke"
      name="star"
      onClick={[Function]}
      size="huge"
      style={
        Object {
          "color": "whitesmoke",
          "position": "absolute",
          "right": 0,
          "top": 9,
          "zIndex": 1,
        }
      }
    />
    <Image
      as="img"
      ui={true}
    />
    <CardContent>
      <CardHeader>
        The Sealed Portion
      </CardHeader>
    </CardContent>
    <CardContent
      extra={true}
    >
      <p>
        <Icon
          as="i"
          name="file text outline"
        />
        22 hr read
      </p>
    </CardContent>
  </Card>
</div>
`;
